sys/linux/socket.txt

```diff
@@ -317,6 +321,41 @@ ifr_ifru [
 
 define IFREQ_SIZE	sizeof(struct ifreq)
 
+# https://elixir.bootlin.com/linux/v5.3.11/source/include/uapi/linux/if_vlan.h
+vlan_args {
+	cmd		flags[vlan_cmd, intptr]
+	device1		array[int8, 24]
+	u		vlan_u
+	vlan_qos	int8
```

Comment:
s/int8/int16/
These types are crucial, they control layout of data in memory. If one of them is incorrect, all data is shifted and kernel reads not what we are passing and nothing is working.

Reply:
Really sorry for this mistake, I have go over the other definitions to make sure everything looks good now.